probe-rs/src/session.rs

```diff
@@ -138,40 +120,120 @@ impl Session {
 
         let mut session = match target.architecture() {
             Architecture::Arm => {
-                if !probe.has_arm_interface() {
-                    return Err(anyhow!(
-                        "Debugging ARM based chips is not supported with the connected probe."
-                    )
-                    .into());
-                }
+                let config = target.cores[0].clone();
+                let arm_core_access_options = match config.core_access_options {
+                    probe_rs_target::CoreAccessOptions::Arm(opt) => Ok(opt),
+                    probe_rs_target::CoreAccessOptions::Riscv(_) => {
+                        Err(AccessPortError::InvalidCoreAccessOption(config.clone()))
+                    }
+                }?;
 
-                let interface = probe.try_into_arm_interface().map_err(|(_, err)| err)?;
+                let default_memory_ap = MemoryAp::new(ApAddress {
+                    dp: match arm_core_access_options.psel {
+                        0 => DpAddress::Default,
+                        x => DpAddress::Multidrop(x),
+                    },
+                    ap: arm_core_access_options.ap,
+                });
 
-                let mut session = Session {
-                    target,
-                    interface: ArchitectureInterface::Arm(interface),
-                    cores,
+                let sequence_handle = match &target.debug_sequence {
+                    DebugSequence::Arm(sequence) => sequence.clone(),
+                    DebugSequence::Riscv => {
+                        panic!("Mismatch between architecture and sequence type!")
+                    }
                 };
 
-                // Todo: Add multicore support. How to deal with any cores that are not active and won't respond?
-                // Enable debug mode
-                debug_core_start(&mut session.core(0)?)?;
+                if AttachMethod::UnderReset == attach_method {
+                    if let Some(dap_probe) = probe.try_as_dap_probe() {
+                        sequence_handle.reset_hardware_assert(dap_probe)?;
+                    } else {
+                        log::info!(
+                            "Custom reset sequences are not supported on {}.",
+                            probe.get_name()
+                        );
```

Comment:
I thought about this. The problem is that currently, with the trait based system, we can only replace none or all sequence handlers. Which means that if another sequence handler is required, but not the reset one, this wont work, even tho it would never be an issue with the actual hardware but a limitation of our API.
So ultimatley we need a better API for this. I will write a bit more down below as I think @Dirbaio made similar remarks.